### ISSUE TITLE
Server Startup Error - 'Couldn't find file for package' [Cosmetic]

### ISSUE THREAD

**Issue by Subressor** [truong-bui/AsyncLoadingScreen#24]
After installing the Plugin (great work btw) and doing vanilla config, package and build check, there are errors on startup that reference 'MagicLeap' and 'OculusVR' content.

![image](https://user-images.githubusercontent.com/40392247/119265091-6022ec00-bbdd-11eb-80dd-472d4c53be05.png)

Not sure where these are referenced in the code. I expect this is because I package my project without editor content

**Comment by professionaljones** [comment]
Are those plugins enabled in the project? They might be by default and removing them may solve the issue

**Comment by truong-bui** [comment]
Hi, this isn't related to Async Loading Screen, I think you should disable the 'MagicLeap' and 'OculusVR' plugin in Plugins settings

**Comment by Subressor** [comment]
Thank you for the reply @truong-bui 

I have disabled the MagicLeap & Oculus Plugins and the last 2 errors have indeed gone. However, the first error still remains. This is the font that I used in the project settings ASync page for all font options available. Changing the font here also changes the error. Can you please advise? Disabling the ASync plugin removes the error.

I have tried adding the entire folder where I keep all of my fonts are included in the package (in packing settings). In-game, the font appears normally.

**Comment by truong-bui** [comment]
@Subressor Open **Settings > Packaging > Packaging** then add your font folder in **Additional Aset Directories to Cook** array
![image](https://user-images.githubusercontent.com/1840964/119314067-79d83800-bc9e-11eb-9f5c-d34de89f34dd.png)

**Comment by Subressor** [comment]
@truong-bui Hey,

Yes that is what I have already tried (similar experience with GAS)

![image](https://user-images.githubusercontent.com/40392247/119315209-86906800-bc6d-11eb-9f55-65d7682eb90d.png)

However the error persists between builds. Is it possible that setting every font family (regardless if its used or not) could be causing this?

![image](https://user-images.githubusercontent.com/40392247/119315364-aaec4480-bc6d-11eb-9ff7-de8bfb585e37.png)

**Comment by truong-bui** [comment]
Try to use default font then build if this issue still happens. I'll investigate this later

**Comment by Subressor** [comment]
Thanks for your support :-)

I tried changing half back to the default (Roboto) and left half with the overlock font, and I now get both errors when running my packaged game

![image](https://user-images.githubusercontent.com/40392247/119372813-a9dc0700-bcaf-11eb-90ae-513318354448.png)

Let me try doing it with just Roboto and report back

**Comment by Subressor** [comment]
Yeap, same issue, just Roboto font erroring now

![image](https://user-images.githubusercontent.com/40392247/119464954-2667f780-bd3b-11eb-874f-a9d11c1a0236.png)

Build Settings for convenience:

**Comment by truong-bui** [comment]
@Subressor It's weird. Try to create an empty project with Async Loading Screen, and add your custom font, then build to see if you will be able to package the project. If yes, I think this is your project issue, not the plugin. Also, I have no problem packaging with custom fonts:
![image](https://user-images.githubusercontent.com/1840964/119508059-9c9b4700-bd99-11eb-9af2-b218253261ad.png)
![image](https://user-images.githubusercontent.com/1840964/119508987-6d390a00-bd9a-11eb-801e-1ea9443d1335.png)

**Comment by Subressor** [comment]
Ah, I think I might have found it, apologies for missing out the detail:
This error is only generated on the dedicated server. I double checked and don't get this error on the client from my side.

Tested in a blank project and getting these errors on basic server build. Can you see them from your side?

**Comment by truong-bui** [comment]
Okay, glad that you found it. I think this issue isn't actually related to my plugin at all, so I'll close this, thanks.

**Comment by Subressor** [comment]
@truong-bui It is directly related to your plugin, because if I disable the plugin the error dissapears. I guess some code is needed to only make this relevant to be packaged/referenced on non-server builds?

When built, the plugin appears in the final packaged server bundle, including a content folder

**Comment by truong-bui** [comment]
I think you should disable this plugin for server build cause it doesn't suppose to use in server-side

**Comment by Subressor** [comment]
Thank you for your continued responses.

I don't think its possible to disable a plugin for server build unless it's in the C++ code of the plugin itself? I can see in your code you have dedicated sections and checks for if its playing on a dedicated server. Perhaps this could be reviewed? For example, in your Startup Module code:

![image](https://user-images.githubusercontent.com/40392247/119888438-42cb8600-bf2d-11eb-870d-fc359d3b08e8.png)

And here also

![image](https://user-images.githubusercontent.com/40392247/119888326-20d20380-bf2d-11eb-8d1f-2e2e3635cfd7.png)

**Comment by truong-bui** [comment]
This code just prevents MoviePlayer plays on the server, because the plugin doesn't suppose to be used on the server-side. You're better to [set up a server build target](https://docs.unrealengine.com/4.26/en-US/InteractiveExperiences/Networking/HowTo/DedicatedServers/). And add the plugin is DisablePlugin array in the server build target file, like in this [answer](https://answers.unrealengine.com/questions/202262/disabling-plugins-for-specific-build-configuration.html) suggests

**Comment by Subressor** [comment]
Hey, thank you for the reply and link!

I have updated my build server target as per your guidance using the links. Unfortunately there is a similar (but new) error:
![image](https://user-images.githubusercontent.com/40392247/119971783-54eb0a00-bfa9-11eb-8fe0-691be197ff92.png)

My Build Settings updated:
![image](https://user-images.githubusercontent.com/40392247/119972002-967bb500-bfa9-11eb-97b0-5b28bf7c9b7d.png)

It seems like the GameInstance spawned on the server is looking for the Plugin, which is now forcibly excluded from the build (which I guess is the root for why it's looking for the font).

Any guidance on the above?

**Comment by truong-bui** [comment]
You should use the **IsServer** node in the GameInstance to prevent the code runs on the server side.

**Comment by Subressor** [comment]
Yeah good shout. I tried that, still gave me an error. I moved the Stop Loading Screen Function out to a widget (which don't load on servers), still gave me an error on the server build.

![image](https://user-images.githubusercontent.com/40392247/120016935-22f39b00-bfdd-11eb-99c4-aa76bb62cb13.png)

It might be related to just that function (Stop Loading Screen). I am not calling it in my code, why is it erroring?

![image](https://user-images.githubusercontent.com/40392247/120017130-66e6a000-bfdd-11eb-9440-2bee403843c3.png)

**Comment by truong-bui** [comment]
I think the best solution in your case is to disable this plugin completely in Plugins when you build for the server and enable it for the client. My networking knowledge is limited, so I couldn't help you more, sorry.

**Comment by Subressor** [comment]
Thanks for your reply's and support on this truong.

I've messed around with a lot more settings and I can't find any way to stop the errors appearing on the server. It doesn't affect the functionality from what I can see, but it is annoying and unclean for the builds going forward.

This doesn't seem to happen with other loading screen implementations, so I guess there may be a way somewhere (whilst still utilizing the ASync loading system) to prevent this from happening.

I would appreciate if you could keep the issue open, in case anyone else comes across this and knows of a way to fix or suppress this warning. I will change the title to reflect the actual issue.

To any future thread reader - ready and happy to implement any suggestions to fixing this, no matter how many years down the line it may be!

**Comment by irajsb** [comment]
When the issues like this arise best way to fix is to study other official plugins with similar functionality .
For example WebMMedia plugin handles this with :
`
"TargetDenyList": [
				"Server"
			]
`in WebMMedia.uplugin . IDK if its enough to prevent registering assets as missing  though .
